### PR TITLE
Scid

### PR DESCRIPTION
**Pull request by benini**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.benini.scid.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
+app-id: com.github.benini.scid
```
Comment (barthalion) [inline com.github.benini.scid.yml:1]:
```suggestion
app-id: io.github.benini.scid
```

File: com.github.benini.scid.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+branch: stable
```
Comment (barthalion) [inline com.github.benini.scid.yml:2]:
```suggestion
```

File: com.github.benini.scid.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  no-autogen: true
+  make-args:
+  - DYNAMIC_ARCH=1
+  - FC=gfortran
+  - NO_LAPACKE=1
+  - USE_OPENMP=0
+  make-install-args:
+  - PREFIX=/app
+  post-install:
+  - rm /app/lib/*.a
```
Comment (barthalion) [inline com.github.benini.scid.yml:51]:
```suggestion
  post-install:
  - rm /app/lib/*.a
```

I recommend using `cleanup` (per module or top-level) for things like these.

File: com.github.benini.scid.yml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  subdir: "unix"
+  post-install:
+    - chmod 755 /app/lib/libtk*.so
+
+- name: scid # main app
+  buildsystem: cmake-ninja
+  post-install:
+    - install -Dma+r com.github.benini.scid.svg /app/share/icons/hicolor/scalable/apps/com.github.benini.scid.svg
+    - install -Dma+r com.github.benini.scid.desktop /app/share/applications/com.github.benini.scid.desktop
+    - install -Dma+r com.github.benini.scid.appdata.xml /app/share/metainfo/com.github.benini.scid.appdata.xml
```
Comment (barthalion) [inline com.github.benini.scid.yml:111]:
```suggestion
    - install -Dma+r com.github.benini.scid.appdata.xml /app/share/appdata/com.github.benini.scid.appdata.xml
```

Alternatively `/app/share/metainfo/xxx.metainfo.xml`, but no mixing up.

File: com.github.benini.scid.yml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    sha256: ece2b5e06dc7c2121c31649be0effdd645f3df53bcf801623246d75494a866ad
+
+- name: tcl # latest stable tcl, required for scid
+  sources:
+    - type: archive
+      url: https://prdownloads.sourceforge.net/tcl/tcl8.6.13-src.tar.gz
+      sha256: 43a1fae7412f61ff11de2cfd05d28cfc3a73762f354a417c62370a54e2caf066
+  subdir: "unix"
+  post-install:
+    - chmod 755 /app/lib/libtcl*.so
```
Comment (barthalion) [inline com.github.benini.scid.yml:95]: Is this actually needed?

File: com.github.benini.scid.yml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+- name: tk # latest stable tk, required for scid
+  sources:
+    - type: archive
+      url: https://prdownloads.sourceforge.net/tcl/tk8.6.13-src.tar.gz
+      sha256: 2e65fa069a23365440a3c56c556b8673b5e32a283800d8d9b257e3f584ce0675
+  subdir: "unix"
+  post-install:
+    - chmod 755 /app/lib/libtk*.so
```
Comment (barthalion) [inline com.github.benini.scid.yml:104]: Is this actually needed?

File: com.github.benini.scid.yml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    - type: git
+      url: https://github.com/benini/scid.git
+      tag: v4.8.0
+      commit: 43740990c3ee47d662d57d11bedac6e667a06fd8
+    - type: file
+      path: com.github.benini.scid.svg
+    - type: file
+      path: com.github.benini.scid.desktop
+    - type: file
+      path: com.github.benini.scid.appdata.xml
```
Comment (barthalion) [inline com.github.benini.scid.yml:122]: These files were not added, and in general belong to https://github.com/benini/scid.git, not here.

File: com.github.benini.scid.yml (hunk 5 above)
Comment (benini) [reply com.github.benini.scid.yml:95]: Avoid failure when stripping the debug information:
https://github.com/flatpak/flatpak/issues/469

File: com.github.benini.scid.yml (hunk 6 above)
Comment (benini) [reply com.github.benini.scid.yml:104]: Same as above

File: com.github.benini.scid.yml [hunk 8]
```diff
@@ -0,0 +1,122 @@
+app-id: com.github.benini.scid
+branch: stable
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: /app/scid/scid
```
Comment (hfiguiere) [inline com.github.benini.scid.yml:6]: the command should be `/app/bin` and the command property a command name, not a path.

File: com.github.benini.scid.yml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    - install -Dma+r com.github.benini.scid.svg /app/share/icons/hicolor/scalable/apps/com.github.benini.scid.svg
+    - install -Dma+r com.github.benini.scid.desktop /app/share/applications/com.github.benini.scid.desktop
+    - install -Dma+r com.github.benini.scid.appdata.xml /app/share/metainfo/com.github.benini.scid.appdata.xml
+  sources:
+    - type: git
+      url: https://github.com/benini/scid.git
+      tag: v4.8.0
+      commit: 43740990c3ee47d662d57d11bedac6e667a06fd8
+    - type: file
+      path: com.github.benini.scid.svg
```
Comment (hfiguiere) [inline com.github.benini.scid.yml:118]: isn't this in the git repo:

https://github.com/benini/scid/blob/github/resources/svg/scid_app.svg

Comment (benini) [reply com.github.benini.scid.yml:118]: A square icon is required?
https://github.com/benini/scid/commit/68efd7cd19e194cee4aa550e434df9ddfc9fc700

File: com.github.benini.scid.yml (hunk 8 above)
Comment (benini) [reply com.github.benini.scid.yml:6]: The binary need to be in /app/scid

Comment (hfiguiere) [reply com.github.benini.scid.yml:6]: symlink, wrapper script. etc. Many solutions are available to work around broken software.

Comment (benini) [reply com.github.benini.scid.yml:6]: oh, that's rude

File: com.github.benini.scid.yml (hunk 9 above)
Comment (barthalion) [reply com.github.benini.scid.yml:118]: Yes, it needs to be square.

**Comment by barthalion** [comment]
bot, build com.github.benini.scid

File: com.github.benini.scid.yml (hunk 8 above)
Comment (barthalion) [reply com.github.benini.scid.yml:6]: It's not strictly required but definitely a good practice.

File: com.github.benini.scid.yml (hunk 9 above)
Comment (rolandlo) [reply com.github.benini.scid.yml]: Please also note that in v4.8.0 there is no desktop file, no metainfo file and no square icon. Either these 3 files should be added here or a new tag has to be created on the uppstream repo and used here.

**Comment by benini** [comment]
bot, build io.github.benini.scid

**Comment by barthalion** [comment]
bot, build io.github.benini.scid

**Comment by benini** [comment]
It is possible to detect the architecture in the flatpak manifest?
Stockfish builds on x86_64 with just `make -j build` but needs `make ARCH=armv8 -j build` for arm.

**Comment by barthalion** [comment]
```
"build-options" : {
  "arch": {
    "make-args": ["ARCH=armv8"]
  }
}
```

Something along this.

**Comment by rolandlo** [comment]
Compare with https://github.com/flathub/com.github.sakya.corechess/blob/3f9898ae4fb36e0750e3c0eb83ee9543384a8d70/com.github.sakya.corechess.json#L66-L88

**Comment by benini** [comment]
bot, build io.github.benini.scid

**Comment by benini** [comment]
bot, build io.github.benini.scid

**Comment by benini** [comment]
bot, build io.github.benini.scid

**Comment by barthalion** [comment]
bot, build io.github.benini.scid

**Comment by barthalion** [comment]
bot, build io.github.benini.scid

**Pull request closed without merging**